remoteappmanager/db/orm.py

```diff
@@ -156,23 +166,60 @@ def transaction(session):
 def apps_for_user(session, user):
     """Returns a list of tuples, each containing an application and the
     associated policy that the specified orm user is allowed to run.
+    If the user is None, the default is to return an empty list.
 
     Parameters
     ----------
     session : Session
         The current session
     user : User
-        the orm User.
+        the orm User, or None.
 
     Returns
     -------
     A list of tuples (orm.Application, orm.ApplicationPolicy)
     """
 
-    with transaction(session):
-        teams = user.teams
+    if user is None:
+        return []
+
+    teams = user.teams
+
+    res = session.query(Accounting).filter(
+        Accounting.team_id.in_([team.id for team in teams])).all()
+
+    return [(acc.application, acc.application_policy) for acc in res]
+
+
+def user_can_run(session, user, application, policy):
+    """Returns True if the user can run a given application with a specific
+    policy. False otherwise. Note that the user can be None.
+    In that case, returns False.
+
+    Parameters
+    ----------
+    session : Session
+        The current session
+    user : User
+        the orm User, or None.
+    application : Application
+        The application object
+    policy : ApplicationPolicy
+        The application policy
+
+    Returns
+    -------
+    boolean
+
+    """
+    if user is None:
+        return False
+
+    team_ids = [team.id for team in user.teams]
+    res = session.query(Accounting) \
+        .filter(Accounting.team_id.in_(team_ids)) \
+        .filter(Accounting.application == application) \
+        .filter(Accounting.application_policy == policy).count()
```

Comment:
`exists` would be more efficient that `count`
http://docs.sqlalchemy.org/en/latest/orm/query.html?highlight=exists#sqlalchemy.orm.query.Query.exists